core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -2231,6 +2241,77 @@ private void updateStringValuesToDatabase(Connection dbConnection, String sqlStm
         }
     }
 
+    private void updateValuesToDatabaseWithUTCTime(Connection dbConnection, String sqlStmt, Object... params)
+            throws UserStoreException {
+
+        PreparedStatement prepStmt = null;
+        boolean localConnection = false;
+        try {
```

Comment:
Use try with resources

Reply:
Can not use try with resource here since we need a final block to close the connections according to a condition